packages/cozy-client/src/Query.jsx

```diff
@@ -72,30 +114,18 @@ export default class Query extends Component {
     }
   }
 
-  recomputeChildrenArgs() {
-    const query = this.observableQuery
-    this.data = {
-      fetchMore: this.fetchMore,
-      fetch: this.fetch,
-      ...query.currentResult()
-    }
-    this.methods = {
-      createDocument: this.createDocument,
-      saveDocument: this.saveDocument,
-      deleteDocument: this.deleteDocument,
-      getAssociation: this.getAssociation,
-      ...this.mutations
-    }
-  }
-
   onQueryChange = () => {
     this.recomputeChildrenArgs()
     this.setState(dummyState)
   }
 
+  recomputeChildrenArgs() {
+    this.childrenArgs = computeChildrenArgs(this)
+  }
+
   render() {
     const { children } = this.props
-    return children(this.data, this.methods)
+    return children(this.childrenArgs[0], this.childrenArgs[1])
```

Comment:
I wonder if using `return children(...this.childrenArgs)` can be good here. So if we add something to childrenArgs, we don't have to think about adding it here.

If `this.childrenArgs` doesn't change over time, this should always return the same object, even if this is a spread.

Reply:
Seen together, the overhead of using f(...args) seems too much considering that the number of arguments of the render prop will not change.